src/Miunie.Storage/JsonDataStorage.cs

```diff
@@ -14,56 +14,50 @@ public class JsonDataStorage : IDataStorage
 
         public JsonDataStorage()
         {
-            var resourcesDirectory = Directory
-                .CreateDirectory(_resourcesFolder);
+            Directory.CreateDirectory(_resourcesFolder);
         }
 
         public JsonDataStorage(string resourcesFolder)
         {
             _resourcesFolder = resourcesFolder;
-            Directory.CreateDirectory(_resourcesFolder);
+            Directory.CreateDirectory(resourcesFolder);
         }
 
         public void StoreObject(object obj, string collection, string key)
         {
-            var file = GetFileNameByKey(key);
-            EnsureCollectionExists(collection);
-
-            string json = JsonConvert
-                .SerializeObject(obj, Formatting.Indented);
-
-            string filePath = Path.Combine(_resourcesFolder, collection, file);
+            EnsureCollectionDirectoryExists(collection);
+            var filePath = GetFullFilePath(collection, key);
+            var json = JsonConvert.SerializeObject(obj);
             File.WriteAllText(filePath, json);
         }
 
         public T RestoreObject<T>(string collection, string key)
         {
-            var file = GetFileNameByKey(key);
-            EnsureCollectionExists(collection);
-            var filePath = String.Concat(collection, "/", file);
-            return RestoreByPath<T>(filePath);
+            EnsureCollectionDirectoryExists(collection);
+            var filePath = GetFullFilePath(collection, key);
+            var json = GetOrCreateFileContent(filePath);
+            return JsonConvert.DeserializeObject<T>(json);
         }
 
         public IEnumerable<T> RestoreCollection<T>(string collection)
         {
-            EnsureCollectionExists(collection);
-            var collectionPath = Path.Combine(_resourcesFolder, collection);
-            var filePaths =  Directory.GetFiles(collectionPath);
-            var files = new HashSet<T>();
-            foreach (var filePath in filePaths)
-            {
-                var fileName = Path.GetFileName(filePath);
-                var file = RestoreObject<T>(collection, fileName);
-                files.Add(file);
-            }
-            return files;
+            EnsureCollectionDirectoryExists(collection);
+            var collectionDir = GetCollectionDirectory(collection);
+            IEnumerable<string> fileKeys = Directory.GetFiles(collectionDir).Select(Path.GetFileNameWithoutExtension);

```

Comment:
When "new-lining" a chain, put each link on a new line.

:x: Don't :
```cs
var results = collection.Select(foo)
    .Where(bar);
```

:heavy_check_mark: Do :
```cs
var results = collection
    .Select(foo)
    .Where(bar);
```

This could be reasoned about as part of the `Make similar code look similar.`.